lua/visual.lua

```diff
@@ -199,8 +201,11 @@ core.register_entity("drawers:visual", {
 			if clicker:get_wielded_item():get_count() > leftover:get_count() then
 				inventoryChanged = true
 			end
-			-- set the leftover as new wielded item for the player
-			clicker:set_wielded_item(leftover)
+			-- set the leftover as new wielded item for the player if not creative
+			if not (minetest.check_player_privs(clicker, "creative") or
+				minetest.is_creative_enabled("")) then
+					clicker:set_wielded_item(leftover)
```

Comment:
Same as above, with `minetestis_creative_enabled`. Also bad indent levels in both cases.

Reply:
Done